src/to-have-style.js

```diff
@@ -1,20 +1,17 @@
+import { StyleSheet } from 'react-native';
 import { matcherHint } from 'jest-matcher-utils';
 import { diff } from 'jest-diff';
 import chalk from 'chalk';
-import { checkReactElement, mergeAll } from './utils';
+import { checkReactElement } from './utils';
 
 function isSubset(expected, received) {
   return Object.entries(expected).every(([prop, value]) =>
     Array.isArray(value)
-      ? isSubset(mergeAll(value), mergeAll(received[prop]))
-      : received[prop] === value,
+      ? isSubset(StyleSheet.flatten(value), StyleSheet.flatten(received[prop]))
```

Comment:
I am not sure we should call `StyleSheet.flatten()` here, as both styles passed to `isSubset` should be already flat style objects due to call-site using `StyleSheet.flatten()` on each of them.

The values are here look like the values of keys of given styles, e.g. `transform: [{ scale: 2 }, { rotate: '45deg' }],` => `transform: { scale: 2, rotate: '45deg' },`. I am not sure we should flatten them as well, as e.g. in case of`transform` the order of transformations seem to be important and applying the same transformation but in different order can result in different style. See: https://reactnative.dev/docs/transforms#transform

It might be that the original `isSubset` logic is flawed, as it also used `mergeAll()`, and recursively called `isSubset`.

Reply:
So that list of style props that actually have array value seems pretty short:
- [`transform`](https://github.com/facebook/react-native/blob/82ceb8c3e34560340044ef864ec2fda674b1ba8e/Libraries/StyleSheet/private/_TransformStyle.js) - array of transform object with single property
- [`fontVariant`](https://github.com/facebook/react-native/blob/82ceb8c3e34560340044ef864ec2fda674b1ba8e/Libraries/StyleSheet/StyleSheetTypes.js#L588) - array of string literals

As far I understand the order is important for `transform`. While `fontVariant` won't work with neither `mergeAll` nor `StyleSheet.flatten()` as the values here are neither objects nor arrays, which will actually make them return `undefined` (see [here](https://github.com/facebook/react-native/blob/82ceb8c3e34560340044ef864ec2fda674b1ba8e/Libraries/StyleSheet/flattenStyle.js#L19))

Which if I understand the use-case correctly should mean that `isSubset` should probably look like this:
```ts
function isSubset(expected, received) {
  return Object.entries(expected).every(([prop, value]) =>
      this.equals(received?.[prop], value),
  );
}
```

Note that I used [`equals`](https://jestjs.io/docs/expect#thisequalsa-b), Jest helper for deep equality, which might require that we inline the modified `isSubset` implementation into returned objects' `pass` property, as this is where `this.equals` should work correctly:

```ts
return {
    pass: Object.entries(expected).every(([prop, value]) => this.equals(received?.[prop], value))
    message: () => {
      const matcher = `${this.isNot ? '.not' : ''}.toHaveStyle`;
      return [matcherHint(matcher, 'element', ''), expectedDiff(expected, received)].join('\n\n');
    },
  };
```

Usage of deep equality instead of `===` operator is important, otherwise the comparison of two different array literals with equivalent content will fail, because they are `equal` but are not the same object.

Reply:
That's reasonable and it makes sense for me, but with that being said we need to come with a decision on how should we handle styles being `transform: [{ scale: 2 }, { rotate: '45deg' }]` and tests trying to match `toHaveStyle({ transform: [{ rotate: '45deg' }] })`.

By using a deep equality function (like `this.equals`) the above would fail - I thought that was the reason we had a custom `isSubset` fn. I tried to keep the same behavior as we had before using `mergeAll`, not sure if it's another side effect introduced by removing `ramda`.

Thoughts @mdjastrzebski?

Reply:
Since we have only two array properties, `transform` and `fontVariant` I think it makes sense to require full deep equality and not being subset.

For `transform` matching only part of the transform makes little sense, as you can add more transform steps and have your test still past, while the result could be much different.

For `fontVariant` matching only part could make sense, but it's quite rare style, and requiring user to adding additional values is not a big burder.

I think that `isSubset` name was primarily meant for top-level subset relation, that if we ask for certain style prop, e.g. `height` and `fontSize` it is there, while there may also be some other style props.

Reply:
@mdjastrzebski that all makes sense, thanks for the guidance here!

I just pushed suggested changes - please note that in a manner of having the right diff I removed the recursiveness on that `narrow` function (not a big deal as it already receives the result of `StyleSheet.flatten`) and also updated existing test snapshots to respect the full deep equality rule from now on